kafka-client/src/main/java/dev/responsive/kafka/internal/metrics/ResponsiveMetrics.java

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright 2023 Responsive Computing, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package dev.responsive.kafka.internal.metrics;
+
+import java.io.Closeable;
+import java.util.HashMap;
+import java.util.Map;
+import org.apache.kafka.common.MetricName;
+import org.apache.kafka.common.TopicPartition;
+import org.apache.kafka.common.metrics.KafkaMetric;
+import org.apache.kafka.common.metrics.MetricValueProvider;
+import org.apache.kafka.common.metrics.Metrics;
+import org.apache.kafka.common.metrics.Sensor;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@SuppressWarnings("checkstyle:linelength")
+public class ResponsiveMetrics implements Closeable {
+  private static final Logger LOG = LoggerFactory.getLogger(ResponsiveMetrics.class);
+
+  public static final String RESPONSIVE_METRICS_NAMESPACE = "dev.responsive";
+
+  public static final String STORE_METRIC_GROUP = "store-metrics";
+
+  public static final String RESTORE_LATENCY = "restore-latency";
+  public static final String RESTORE_LATENCY_AVG = RESTORE_LATENCY + "-avg";
+  public static final String RESTORE_LATENCY_MAX = RESTORE_LATENCY + "-max";
+  public static final String RESTORE_LATENCY_DESCRIPTION = "amount of time spent restoring this state store before resuming processing";
+
+  private final Metrics metrics;
+  private final Map<String, String> baseTags = new HashMap<>();
+
+  public ResponsiveMetrics(final Metrics metrics) {
+    this.metrics = metrics;
+  }
+
+  /**
+   * @param applicationId the Streams application id, ie the shared consumer group id
+   * @param streamsClientId the Streams client id, ie the process-specific id
+   * @param userTags optional custom tags that will be attached to each metric
+   */
+  public void initialize(
+      final String applicationId,
+      final String streamsClientId,
+      final ClientVersionMetadata versionMetadata,
+      final Map<String, ?> userTags
+  ) {
+    // this is technically redundant with the application id, but sometimes it's useful to have both
+    baseTags.put("consumer-group", applicationId);
+    baseTags.put("application-id", applicationId);
```

Comment:
can we prefix this with "streams". We have application/applicationId tags that we use in other places to refer to the responsive application ID

Reply:
ah yeah good call